### PR TITLE
Added table consistencty for alerts and 2 new sub-options under alert for :- properties and history 

### PR DESCRIPTION
**Pull request by Divyanshu11011**
# Description
Summarize the change.

Fixes #<issue-number> (link all the GitHub issues this addresses)

# Testing
Describe how you tested this code. How can the reviewers reproduce your tests?

# Checklist:
Before marking your pull request as ready for review, complete the following.

- [x] I have self-reviewed this PR.
- [x] I have removed all print-debugging and commented-out code that should not be merged.
- [x] I have added sufficient comments in my code, particularly in hard-to-understand areas.
- [x] I have formatted the code, if applicable. For Go, I have run `goimports -w .`.

### CODE REVIEW

**Comment by Divyanshu11011** [comment]
Added all the 3 functionalities that were described to me !
Below are the snapshots of the implementation :- 
image
image

**Comment by nkunal** [comment]
looks like you deleted the alerts.js and nothing is working, I get blank tables:

image

**Comment by Divyanshu11011** [comment]
alerts.js is still there in my repo as well as my commit , let me check

**Comment by Divyanshu11011** [comment]
Please check now @nkunal sir

**Comment by nkunal** [comment]
one existing functionality broke with this PR:
1) click on creating a new alert (or click edit of an existing alert)
2) then click cancel, it should take you back to the previous list of all-allerts table. (you can verify this functionality on http://playgorund.siglens.com:5122

**Comment by Divyanshu11011** [comment]
On it !

**Comment by Divyanshu11011** [comment]
Have fixed the cancel-btn there are some errors in creating a new alert , fixing that

**Comment by Divyanshu11011** [comment]
edit functionality is left to be fixed ! doing that quickly

**Comment by Divyanshu11011** [comment]
Any guidance on how I can fix this, @nkunal? I have fixed everything else!

**Comment by Divyanshu11011** [comment]
I just clicked on "Update Branch" and test cases got passed ! @nkunal

File: static/css/siglens.css [hunk 1]
```diff
@@ -17,7 +17,7 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-:root {
+ :root {
```
Comment (nkunal) [inline static/css/siglens.css:20]: remove this white space change

File: static/css/siglens.css [hunk 2]
```diff
@@ -4247,6 +4247,8 @@ color: #C8C7CC;
     font-size: 14px;
 }
 
+
+
```
Comment (nkunal) [inline static/css/siglens.css:4251]: remove this white space change

File: static/js/alert.js [hunk 3]
```diff
@@ -43,88 +65,186 @@ function inDoubleBrackets(str) {
...
+        };
+        new agGrid.Grid(eGridDiv, gridOptions);
+    } else {
+        console.error('Element #HistoryDetailsGrid not found');
+    }
+}
+
 function getAlertId() {
     const urlParams = new URLSearchParams(window.location.search);
+    console.log('Getting Alert ID from URL params');
```
Comment (nkunal) [inline static/js/alert.js:220]: remove log

File: static/js/alert.js [hunk 4]
```diff
@@ -43,88 +65,186 @@ function inDoubleBrackets(str) {
...
 
     if (urlParams.has('id')) {
         const id = urlParams.get('id');
+        console.log('Editing Alert with ID:', id);
```
Comment (nkunal) [inline static/js/alert.js:224]: remove log

File: static/js/alert.js [hunk 5]
```diff
@@ -43,88 +65,186 @@ function inDoubleBrackets(str) {
...
         editAlert(id);
         alertID = id;
     } else if (urlParams.has('queryLanguage')) {
         const queryLanguage = urlParams.get('queryLanguage');
         const searchText = urlParams.get('searchText');
         const startEpoch = urlParams.get('startEpoch');
         const endEpoch = urlParams.get('endEpoch');
-    
+
+        console.log('Creating Alert from Logs with Query Language:', queryLanguage);
```
Comment (nkunal) [inline static/js/alert.js:233]: remobve log

File: static/js/alert.js [hunk 6]
```diff
@@ -43,88 +65,186 @@ function inDoubleBrackets(str) {
...
         createAlertFromLogs(queryLanguage, searchText, startEpoch, endEpoch);
+    } else {
+        console.log('No alert ID or query language found in URL params');
```
Comment (nkunal) [inline static/js/alert.js:236]: remove log

File: static/js/alert.js [hunk 7]
```diff
@@ -43,88 +65,186 @@ function inDoubleBrackets(str) {
...
     }
 }
 
-function editAlert(alertId){
+
+
+
+
+
+
```
Comment (nkunal) [inline static/js/alert.js:245]: remove this white space change

File: static/js/alert.js [hunk 8]
```diff
@@ -43,88 +65,186 @@ function inDoubleBrackets(str) {
...
+function editAlert(alertId) {
+    console.log('Fetching alert details for ID:', alertId);
```
Comment (nkunal) [inline static/js/alert.js:247]: remove log

File: static/js/alert.js [hunk 9]
```diff
@@ -135,20 +255,89 @@ function editAlert(alertId){
...
             displayAlert(res.alert);
         }
-    })
+    }).catch(function (error) {
+        console.error('Error fetching alert details:', error);
+    });
 }
 
+function displayAlert(res) {
+    console.log('Displaying alert data');
```
Comment (nkunal) [inline static/js/alert.js:273]: remove log

File: static/js/alert.js [hunk 10]
```diff
@@ -424,20 +591,87 @@ function alertDetailsFunctions(){
...
+        getAlertId();
+    });
+
+    $(document).ready(function() {
+        if (window.location.href.includes("alert-details.html")) {
+            const eGridDiv = document.querySelector('#AlertDetailsGrid');
+            if (eGridDiv) {
+                try {
+                    new agGrid.Grid(eGridDiv, alertGridOptions);
+                    console.log('ag-Grid initialized successfully.');
```
Comment (nkunal) [inline static/js/alert.js:663]: remove log

File: static/js/alert.js [hunk 11]
```diff
@@ -424,20 +591,87 @@ function alertDetailsFunctions(){
...
+                } catch (error) {
+                    console.error('Failed to initialize ag-Grid:', error);
+                }
+            } else {
+                console.error('Element #AlertDetailsGrid not found');
+            }
+        } else {
+            console.log('Not on alert-details.html, skipping grid initialization.');
```
Comment (nkunal) [inline static/js/alert.js:671]: remove log

File: static/js/alert.js [hunk 12]
```diff
@@ -424,20 +591,87 @@ function alertDetailsFunctions(){
...
+        }
+    });
+    
+    
+
```
Comment (nkunal) [inline static/js/alert.js:676]: remove white space

**Review by nkunal — COMMENTED** [review]
1) there are lot of changes in this PR that are not related to this PR, making it very hard to review it. Please redo this PR and only make the changes that are needed.

2) Never keep console.log lines that you added for debug, produces unneccasry logging on the front end

3) remove all unneccassary white space changes

**Pull request closed without merging**